@W-19444570 Add Payment Instrument

syadupathi-sf opened the pull request:
<!--- Provide a short summary of your changes in the Title field above -->

# Description

From the My Account Page, we have the ability to add a new saved payment now

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [x] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

# Changes
<img width="962" height="606" alt="Screenshot 2025-09-09 at 6 00 42 PM" src="https://github.com/user-attachments/assets/894c3c46-f2d4-4c78-86ee-9a8cfb5095bd" />

# How to Test-Drive This PR

- As a logged in user, add a payment
- Verify the payment cards are listed as expected

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [x] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

syadupathi-sf committed:
W-19444570 add payment instrument

cc-prodsec commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/4a07f3ab-695c-4cbe-8aae-861f14238cc3?refs=67fa24b7cd5fd46614618adb10b43758da361977&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/4a07f3ab-695c-4cbe-8aae-861f14238cc3/license?refs=67fa24b7cd5fd46614618adb10b43758da361977&source=prChecksComment)

dannyphan2000 reviewed (approved):
LGTM

vmarta committed:
Make sure create-mobify-app-dev.js works as intended

Bumping up the -dev versions make sure that all packages have versions that have NOT been published to npm yet. Previously we accidentally published `3.12.0-dev` to npm.
